src/functional/fencil_processors/source_modules/source_modules.py

```diff
@@ -46,14 +72,14 @@ class LibraryDependency:
 
 
 @dataclass(frozen=True)
-class SourceModule:
+class SourceModule(Generic[LanguageT]):
     entry_point: Function
     source_code: str
-    library_deps: Sequence[LibraryDependency]
-    language: str
+    library_deps: tuple[LibraryDependency, ...]
+    language: LanguageT
 
 
 @dataclass(frozen=True)
 class BindingModule:
```

Comment:
If `SourceModule` is generic with respect to a `LanguageT` typevar, shouldn't `BindingModule` be generic with respect to the target language, and maybe even also with respect to the `SourceModule` language (contravariantly)?

Reply:
Probably should be 👍